tpgtools/templates/provider_dcl_endpoints.go.tmpl

```diff
@@ -47,24 +50,29 @@ var {{$pkg.BasePathIdentifier.ToTitle}}EndpointEntry = &schema.Schema{
 {{- end}}
 {{end}}
 
-//Add new values to config.go.erb config object declaration
+type DCLConfig struct {
 {{- range $index, $pkg := . }}
 {{- if $pkg.ShouldWriteProductBasePath }}
-//{{$pkg.BasePathIdentifier.ToTitle}}BasePath string
+	{{$pkg.BasePathIdentifier.ToTitle}}BasePath string
 {{- end}}
 {{- end}}
+}
 
-//Add new values to provider.go.erb schema initialization
+func configureDCLProvider(provider *schema.Provider) {
 {{- range $index, $pkg := . }}
 {{- if $pkg.ShouldWriteProductBasePath }}
-// {{$pkg.BasePathIdentifier.ToTitle}}EndpointEntryKey:               {{$pkg.BasePathIdentifier.ToTitle}}EndpointEntry,
+	provider.Schema[{{$pkg.BasePathIdentifier.ToTitle}}EndpointEntryKey] = {{$pkg.BasePathIdentifier.ToTitle}}EndpointEntry
 {{- end}}
 {{- end}}
+}
 
-//Add new values to provider.go.erb - provider block read
+func providerDCLConfigure(ctx context.Context, d *schema.ResourceData, p *schema.Provider, config *Config) (interface{}, diag.Diagnostics) {
```

Comment:
```suggestion
func providerDCLConfigure(d *schema.ResourceData, config *Config) interface{} {
```
recommend not complicating the interface of this function to match the return parameters of the provider.go. If we have no need for diagnostics lets omit it from the return parameters. Same with the function parameters. If not needed, omit.